Hi,
I get an error when running the test. I have all dependencies installed and didn't saw a error while building.

Any ideas?

My system:
openSUSE 12.3
gcc version 4.7.2 20130108
zlib-devel - 1.2.7-7.1.1
libbz2-1 - 1.0.6-23.1.3
libbz2-devel - 1.0.6-23.1.3
libsnappy1  - 1.1.1-8.1
snappy-devel - 1.1.1-9.2
Latests gflags from github, compiled and installed without problems (https://github.com/schuhschuh/gflags)

git clone https://github.com/facebook/rocksdb.git
make clean; make

"make check" output:

python tools/ldb_test.py
Running testCountDelimDump...
ERunning testCountDelimIDump...
ERunning testDumpLoad...
ERunning testHexPutGet...
ERunning testInvalidCmdLines...
.Running testMiscAdminTask...
ERunning testSimpleStringPutGet...
ERunning testStringBatchPut...
ERunning testTtlPutGet...
# E
## ERROR: testCountDelimDump (**main**.LDBTestCase)

Traceback (most recent call last):
  File "tools/ldb_test.py", line 150, in testCountDelimDump
    self.assertRunOK("batchput x.1 x1 --create_if_missing", "OK")
  File "tools/ldb_test.py", line 82, in assertRunOK
    expectedOutput, unexpected)
  File "tools/ldb_test.py", line 54, in assertRunOKFull
    params, shell=True)
  File "tools/ldb_test.py", line 25, in my_check_output
    (retcode, cmd))
Exception: Exit code is not 0.  It is 1.  Command: ./ldb --db=/tmp/ldb_test_r827wv/testdb batchput x.1 x1 --create_if_missing |grep -v "Created bg thread"
# 
## ERROR: testCountDelimIDump (**main**.LDBTestCase)

Traceback (most recent call last):
  File "tools/ldb_test.py", line 159, in testCountDelimIDump
    self.assertRunOK("batchput x.1 x1 --create_if_missing", "OK")
  File "tools/ldb_test.py", line 82, in assertRunOK
    expectedOutput, unexpected)
  File "tools/ldb_test.py", line 54, in assertRunOKFull
    params, shell=True)
  File "tools/ldb_test.py", line 25, in my_check_output
    (retcode, cmd))
Exception: Exit code is not 0.  It is 1.  Command: ./ldb --db=/tmp/ldb_test_kuZOEm/testdb batchput x.1 x1 --create_if_missing |grep -v "Created bg thread"
# 
## ERROR: testDumpLoad (**main**.LDBTestCase)

Traceback (most recent call last):
  File "tools/ldb_test.py", line 231, in testDumpLoad
    "OK")
  File "tools/ldb_test.py", line 82, in assertRunOK
    expectedOutput, unexpected)
  File "tools/ldb_test.py", line 54, in assertRunOKFull
    params, shell=True)
  File "tools/ldb_test.py", line 25, in my_check_output
    (retcode, cmd))
Exception: Exit code is not 0.  It is 1.  Command: ./ldb --db=/tmp/ldb_test_ZHtyAs/testdb batchput --create_if_missing x1 y1 x2 y2 x3 y3 x4 y4 |grep -v "Created bg thread"
# 
## ERROR: testHexPutGet (**main**.LDBTestCase)

Traceback (most recent call last):
  File "tools/ldb_test.py", line 178, in testHexPutGet
    self.assertRunOK("put a1 b1 --create_if_missing", "OK")
  File "tools/ldb_test.py", line 82, in assertRunOK
    expectedOutput, unexpected)
  File "tools/ldb_test.py", line 54, in assertRunOKFull
    params, shell=True)
  File "tools/ldb_test.py", line 25, in my_check_output
    (retcode, cmd))
Exception: Exit code is not 0.  It is 1.  Command: ./ldb --db=/tmp/ldb_test_34UXEu/testdb put a1 b1 --create_if_missing |grep -v "Created bg thread"
# 
## ERROR: testMiscAdminTask (**main**.LDBTestCase)

Traceback (most recent call last):
  File "tools/ldb_test.py", line 324, in testMiscAdminTask
    "OK")
  File "tools/ldb_test.py", line 82, in assertRunOK
    expectedOutput, unexpected)
  File "tools/ldb_test.py", line 54, in assertRunOKFull
    params, shell=True)
  File "tools/ldb_test.py", line 25, in my_check_output
    (retcode, cmd))
Exception: Exit code is not 0.  It is 1.  Command: ./ldb --db=/tmp/ldb_test__57ZVN/testdb batchput --create_if_missing x1 y1 x2 y2 x3 y3 x4 y4 |grep -v "Created bg thread"
# 
## ERROR: testSimpleStringPutGet (**main**.LDBTestCase)

Traceback (most recent call last):
  File "tools/ldb_test.py", line 93, in testSimpleStringPutGet
    self.assertRunOK("put --create_if_missing x1 y1", "OK")
  File "tools/ldb_test.py", line 82, in assertRunOK
    expectedOutput, unexpected)
  File "tools/ldb_test.py", line 54, in assertRunOKFull
    params, shell=True)
  File "tools/ldb_test.py", line 25, in my_check_output
    (retcode, cmd))
Exception: Exit code is not 0.  It is 1.  Command: ./ldb --db=/tmp/ldb_test_bLX8DS/testdb put --create_if_missing x1 y1 |grep -v "Created bg thread"
# 
## ERROR: testStringBatchPut (**main**.LDBTestCase)

Traceback (most recent call last):
  File "tools/ldb_test.py", line 140, in testStringBatchPut
    self.assertRunOK("batchput x1 y1 --create_if_missing", "OK")
  File "tools/ldb_test.py", line 82, in assertRunOK
    expectedOutput, unexpected)
  File "tools/ldb_test.py", line 54, in assertRunOKFull
    params, shell=True)
  File "tools/ldb_test.py", line 25, in my_check_output
    (retcode, cmd))
Exception: Exit code is not 0.  It is 1.  Command: ./ldb --db=/tmp/ldb_test_tqn8ny/testdb batchput x1 y1 --create_if_missing |grep -v "Created bg thread"
# 
## ERROR: testTtlPutGet (**main**.LDBTestCase)

Traceback (most recent call last):
  File "tools/ldb_test.py", line 207, in testTtlPutGet
    self.assertRunOK("put a1 b1 --ttl --create_if_missing", "OK")
  File "tools/ldb_test.py", line 82, in assertRunOK
    expectedOutput, unexpected)
  File "tools/ldb_test.py", line 54, in assertRunOKFull
    params, shell=True)
  File "tools/ldb_test.py", line 25, in my_check_output
    (retcode, cmd))
Exception: Exit code is not 0.  It is 1.  Command: ./ldb --db=/tmp/ldb_test_AkrY96/testdb put a1 b1 --ttl --create_if_missing |grep -v "Created bg thread"

---

Ran 9 tests in 0.072s

FAILED (errors=8)
make: **\* [ldb_tests] Fehler 1

Can you please run ./db_test and send us the output? Tnx.

./db_test: error while loading shared libraries: libgflags.so.2: cannot open shared object file: No such file or directory

My libgflags.so is in /usr/local/lib64/
I've made a link in /usr/lib64 but that didn't help. Where does it look for this lib?

You can try running `ldd ./db_test` -- it should tell you where your environment is looking for libraries

Hm, I tried again today (without any changes) and now it works..
Thanks for your help.

